### ISSUE TITLE
Red Alert Crashes when starting application

### ISSUE THREAD

**Issue by Tom109123** [OpenRA/OpenRA#17898]
Hardware: HP Probook 6550B
OS: Win10 x64, build 1909, Fresh install
Antivirus: Windows Defender+Avast. 

OpenRA Release:  release-20200202

When I start OpenRa on my newly installed laptop, the application crashes. Log is provided below:

OpenRA engine version release-20200202
Mod Content Manager mod version release-20200202
Date: 2020-04-04 13:35:17Z
Operating System: Windows (Microsoft Windows NT 6.2.9200.0)
Runtime Version: .NET CLR 4.0.30319.42000
Exception of type `System.InvalidOperationException`: Collection was modified; enumeration operation may not execute.
   at System.ThrowHelper.ThrowInvalidOperationException(ExceptionResource resource)
   at System.Collections.Generic.List`1.Enumerator.MoveNextRare()
   at OpenRA.Graphics.ChromeProvider.Initialize(ModData modData)
   at OpenRA.ModData.InitializeLoaders(IReadOnlyFileSystem fileSystem)
   at OpenRA.Game.InitializeMod(String mod, Arguments args)
   at OpenRA.Mods.Common.LoadScreens.BlankLoadScreen.BeforeLoad()
   at OpenRA.Game.InitializeMod(String mod, Arguments args)
   at OpenRA.Game.Initialize(Arguments args)
   at OpenRA.Game.InitializeAndRun(String[] args)
   at OpenRA.WindowsLauncher.RunGame(String[] args)

I reinstalled my laptop because i ran into this error some time ago. Thought it was caused by my OS Install, so i reinstalled the complete OS.

**Comment by abcdefg30** [comment]
https://github.com/OpenRA/OpenRA/issues/17906#issuecomment-610607460 suggests that Avast might cause issues here. Can you try disabling it before installing and starting OpenRA?

**Comment by maroonout32** [comment]
I am receiving this issue as well. I do not have Avast installed on my computer. Fresh install on an older pc. 

OpenRA engine version release-20200202
Mod Content Manager mod version release-20200202
Date: 2020-04-08 14:53:00Z
Operating System: Windows (Microsoft Windows NT 6.2.9200.0)
Runtime Version: .NET CLR 4.0.30319.42000
Exception of type `System.InvalidOperationException`: Collection was modified; enumeration operation may not execute.
   at System.ThrowHelper.ThrowInvalidOperationException(ExceptionResource resource)
   at System.Collections.Generic.List`1.Enumerator.MoveNextRare()
   at OpenRA.Graphics.ChromeProvider.Initialize(ModData modData)
   at OpenRA.ModData.InitializeLoaders(IReadOnlyFileSystem fileSystem)
   at OpenRA.Game.InitializeMod(String mod, Arguments args)
   at OpenRA.Mods.Common.LoadScreens.BlankLoadScreen.BeforeLoad()
   at OpenRA.Game.InitializeMod(String mod, Arguments args)
   at OpenRA.Game.Initialize(Arguments args)
   at OpenRA.Game.InitializeAndRun(String[] args)
   at OpenRA.WindowsLauncher.RunGame(String[] args)

**Comment by Tom109123** [comment]
I have disabled Windows Defender. I have removed Avast. Same story.
Any thoughts / ideas?

**Comment by pchote** [comment]
Does the playtest build crash for you too?

**Comment by Tom109123** [comment]
Yes. it crashes earlier compared to the latest version. In the latest version I Initially see a black screen with the RA Logo.
Crash logs:
OpenRA engine version playtest-20200329
Date: 2020-04-08 16:39:26Z
Operating System: Windows (Microsoft Windows NT 6.2.9200.0)
Runtime Version: .NET CLR 4.0.30319.42000
Exception of type `System.InvalidOperationException`: Failed to initialize platform-integration library. Check graphics.log for details.
   at OpenRA.Game.Initialize(Arguments args)
   at OpenRA.Game.InitializeAndRun(String[] args)
   at OpenRA.WindowsLauncher.RunGame(String[] args)

**Comment by pchote** [comment]
Can you please add the `graphics.log` written by the playtest crash?

That will be an unrelated issue (or maybe not, if as I suspect the underlying bug is in Windows/.NET framework) but if there is an easy solution for that it would be helpful for us to know whether the first crash has already been resolved.

**Comment by Tom109123** [comment]
this file is empty.
However....I somehow managed to fix this issue.
Steps I took, which didn't resolve the issue:
Change regional settings: decimal/digit symbol, changed , with ., and vice versa. Didn't solve the issue.
Checked graphics card driver updates on Intel website,no driver updates for onboard Intel HD Graphics Card. 
I suspected .net related issues too.in control panel, removed 1 .Net Windows update. No fix
I removed .Net 2.5 and 3.5 Windows features, reboot-->No fix.
Ran .net Cleanup Tool, and re-enabled .Net 2.5 and 3.5 Windows feature. Still-->no fix.
After installing playtest version, I noticed Windows Smartscreen notice. I went to App&Browser control, 
i turned these settings to OFF: 
- Check Apps&Files
- Smartscreen for Microsoft Edge
- SmartScreen for Windows Store Apps

Then in Exploit protection settings:
- CFG-->OFF
- DEP--> OFF
- Mandatory ASLR --> OFF

After these 6 settings, rebooted. Issue still there. I enabled all settings again, and rebooted. After this I was about to check for the graphics.log, and noticed it was an empty file. I thought it might work if i started Red Alert and wait for the crash, and then grab the graphics log, however, Red Alert Started!
I think it's one of those last 6 settings which i have re-enabled, which fixed this issue.
So again, my issue is solved! I am running Avast+Windows Defender, these don't cause any issues.

If you would like me to tast any further, please let me know, I am happy to.

**Comment by ghost** [comment]
Hey,
my laptop Asus i3 350M , 4 GB ram.  Windows 10 x64 . Fresh install
Antivirus: Windows Defender+McAfee IS.
OpenRA Release: release-20200202

I am starting the game and there is a problem:
My log:
OpenRA engine version release-20200202
Mod Content Manager mod version release-20200202
Date: 2020-04-09 08:50:35Z
Operating System: Windows (Microsoft Windows NT 6.2.9200.0)
Runtime Version: .NET CLR 4.0.30319.42000
Exception of type `System.InvalidOperationException`: Zmodyfikowano kolekcję. Nie można wykonać operacji wyliczania.
   w System.ThrowHelper.ThrowInvalidOperationException(ExceptionResource resource)
   w System.Collections.Generic.List`1.Enumerator.MoveNextRare()
   w OpenRA.Graphics.ChromeProvider.Initialize(ModData modData)
   w OpenRA.ModData.InitializeLoaders(IReadOnlyFileSystem fileSystem)
   w OpenRA.Game.InitializeMod(String mod, Arguments args)
   w OpenRA.Mods.Common.LoadScreens.BlankLoadScreen.BeforeLoad()
   w OpenRA.Game.InitializeMod(String mod, Arguments args)
   w OpenRA.Game.Initialize(Arguments args)
   w OpenRA.Game.InitializeAndRun(String[] args)
   w OpenRA.WindowsLauncher.RunGame(String[] args)

pref.log
     4 ms LoadMaps

**Comment by pchote** [comment]
@Mariusz1975pl: Does the playtest build crash for you too?

**Comment by ghost** [comment]
From playtest version 

OpenRA engine version playtest-20200329
Date: 2020-04-09 08:59:08Z
Operating System: Windows (Microsoft Windows NT 6.2.9200.0)
Runtime Version: .NET CLR 4.0.30319.42000
Exception of type `System.InvalidOperationException`: Failed to initialize platform-integration library. Check graphics.log for details.
   w OpenRA.Game.Initialize(Arguments args)
   w OpenRA.Game.InitializeAndRun(String[] args)
   w OpenRA.WindowsLauncher.RunGame(String[] args)

graphics.log:

System.InvalidOperationException: Can not create OpenGL context. (Error: Invalid window)
   w OpenRA.Platforms.Default.Sdl2GraphicsContext.InitializeOpenGL()
   w OpenRA.Platforms.Default.Sdl2PlatformWindow..ctor(Size requestEffectiveWindowSize, WindowMode windowMode, Single scaleModifier, Int32 batchSize, Int32 videoDisplay)
   w OpenRA.Platforms.Default.DefaultPlatform.CreateWindow(Size size, WindowMode windowMode, Single scaleModifier, Int32 batchSize, Int32 videoDisplay)
   w OpenRA.Renderer..ctor(IPlatform platform, GraphicSettings graphicSettings)
   w OpenRA.Game.Initialize(Arguments args)
System.InvalidOperationException: Can not create OpenGL context. (Error: Invalid window)
   w OpenRA.Platforms.Default.Sdl2GraphicsContext.InitializeOpenGL()
   w OpenRA.Platforms.Default.Sdl2PlatformWindow..ctor(Size requestEffectiveWindowSize, WindowMode windowMode, Single scaleModifier, Int32 batchSize, Int32 videoDisplay)
   w OpenRA.Platforms.Default.DefaultPlatform.CreateWindow(Size size, WindowMode windowMode, Single scaleModifier, Int32 batchSize, Int32 videoDisplay)
   w OpenRA.Renderer..ctor(IPlatform platform, GraphicSettings graphicSettings)
   w OpenRA.Game.Initialize(Arguments args)

**Comment by maroonout32** [comment]
I've tried uninstalling and reinstalling both stable and playtest releases, and going through the steps above, nothing has worked so far.

**Comment by pchote** [comment]
@maroonout32, @Mariusz1975pl: does the previous [release-20191117](https://github.com/OpenRA/OpenRA/releases/tag/release-20191117) work for you?

If it does, please try reinstalling the playtest, download https://www.libsdl.org/release/SDL2-2.0.5-win32-x64.zip, and copy the SDL2.dll file over the version in the playtest install directory.

**Comment by ghost** [comment]
I install release-20191117 version but it won't start.
OpenRA engine version release-20191117
Mod Content Manager mod version release-20191117
Date: 2020-04-13 15:38:08Z
Operating System: Windows (Microsoft Windows NT 6.2.9200.0)
Runtime Version: .NET CLR 4.0.30319.42000
Exception of type `System.InvalidOperationException`: Zmodyfikowano kolekcję. Nie można wykonać operacji wyliczania.
   w System.ThrowHelper.ThrowInvalidOperationException(ExceptionResource resource)
   w System.Collections.Generic.List`1.Enumerator.MoveNextRare()
   w OpenRA.Graphics.ChromeProvider.Initialize(ModData modData)
   w OpenRA.ModData.InitializeLoaders(IReadOnlyFileSystem fileSystem)
   w OpenRA.Game.InitializeMod(String mod, Arguments args)
   w OpenRA.Mods.Common.LoadScreens.BlankLoadScreen.BeforeLoad()
   w OpenRA.Game.InitializeMod(String mod, Arguments args)
   w OpenRA.Game.Initialize(Arguments args)
   w OpenRA.Game.InitializeAndRun(String[] args)
   w OpenRA.WindowsLauncher.RunGame(String[] args)

5 ms LoadMaps  from pref.log

**Comment by ghost** [comment]
I reinstalled playtest and copy SDL2.dll file but it won't start.

**Comment by JonaVBB** [comment]
Same here, with a completely new Windows10 setup.
The latest stable release-20200503 ran only one time, but the mouse speed was stuttering. I limited the frames to 60. After that the games started crashing on startup continously. The same with the latest playtest-20210131.

OpenRA engine version release-20200503
Date: 2021-02-01 12:21:48Z
Operating System: Windows (Microsoft Windows NT 6.2.9200.0)
Runtime Version: .NET CLR 4.0.30319.42000
Exception of type `System.InvalidOperationException`: Failed to initialize platform-integration library. Check graphics.log for details.
   bei OpenRA.Game.Initialize(Arguments args)
   bei OpenRA.Game.InitializeAndRun(String[] args)
   bei OpenRA.WindowsLauncher.RunGame(String[] args)

The  graphics.log, however, is empty.

I tried the previous release-20191117 as suggested by pchote. This  previous release-20191117 runs stable on my machine.

Hope, this description helps.
